common/static/js/capa/src/formula_equation_preview.js

```diff
@@ -169,30 +169,22 @@ formulaEquationPreview.enable = function() {
         }
 
         function display(latex) {
-            MathJax.Hub.Startup.signal.Interest(function(message) {
-                if (message === 'End') {
+            // Check if MathJax is loaded
+            if (typeof MathJax !== 'undefined' && MathJax !== null) {
+                MathJax.startup.promise.then(() => {
                     var previewElement = inputData.$preview[0];
-                    MathJax.Hub.Queue(function() {
-                        inputData.jax = MathJax.Hub.getAllJax(previewElement)[0];
-                    });
-
-                    MathJax.Hub.Queue(function() {
-                        // Check if MathJax is loaded
-                        if (inputData.jax) {
-                            // Set the text as the latex code, and then update the MathJax.
-                            MathJax.Hub.Queue(
-                                ['Text', inputData.jax, latex]
-                            );
-                        } else if (latex) {
-                            console.log('[FormulaEquationInput] Oops no mathjax for ', latex);
-                            // Fall back to modifying the actual element.
-                            var textNode = previewElement.childNodes[0];
-                            textNode.data = '\\(' + latex + '\\)';
-                            MathJax.Hub.Queue(['Typeset', MathJax.Hub, previewElement]);
+                    // Set the text as the latex code, and then update the MathJax.
+                    MathJax.typesetClear([previewElement]);
+                    // Surround eqn with backticks "`", so that mathjax can process it as asciimath
+                    previewElement.innerHTML = `\`${latex}\``;
```

Comment:
Is there a reason why this is to be processed as asciimath and not as latex as it was previously the case?

Reply:
@CefBoud It was also the case before, AFAIK using back-ticks allow us to use both asciimath and tex commands. In version 3, the raw commands are stored in an external list of math objects while in version 2, it was stored in a `<script>` tag next to the rendered version so it was possible to update the raw command directly without removing elements.

> it is a bit confusing parsing a variable named latex as asciimath.

As mentioned above, it can also parse and render latex commands but `\[` & `\(` delimiters can only parse tex commands.

Reply:
> As mentioned above, it can also parse and render latex commands but \[ & \( delimiters can only parse tex commands.

@navinkarkera  Thank you for the explanation. I understand the reasoning.

Nit: however  if I am not mistaken, the input of the display function can only be latex because it goes through [preview_formcalc](https://github.com/openedx/edx-platform/blob/3bda3bed4ae2007763ce9197ba5258618f95d093/common/static/js/capa/src/formula_equation_preview.js#L125-L126) first which ends up calling [latex_preview ](https://github.com/openedx/edx-platform/blob/3bda3bed4ae2007763ce9197ba5258618f95d093/xmodule/capa/inputtypes.py#L1307), so I am not sure if there is a scenario in which the input of `display`  ends up being an asciimath expression.

Reply:
@CefBoud You are right :100:, I just did not see any disadvantages with supporting both in frontend (maybe we add support for it in backend in future :smiley_cat:). Still, if you feel like we should change it to `\(`, I'll update it.

Reply:
@navinkarkera 
I think it would be best if it's `\(` to avoid any confusion. So I'd advise the change if you don't mind. 🙂

Reply:
@CefBoud Done.